recipes-openxt/qemu-dm/qemu-dm-2.6.2/ovmf-q35.patch

```diff
@@ -0,0 +1,284 @@
+Index: qemu-2.6.2/hw/i386/pc_piix.c
+===================================================================
+--- qemu-2.6.2.orig/hw/i386/pc_piix.c
++++ qemu-2.6.2/hw/i386/pc_piix.c
+@@ -56,6 +56,8 @@
+ #include "migration/migration.h"
+ #include "kvm_i386.h"
+ 
++extern int xen_q35;
++
+ #define MAX_IDE_BUS 2
+ 
+ static const int ide_iobase[MAX_IDE_BUS] = { 0x1f0, 0x170 };
+@@ -377,6 +379,8 @@ static void pc_xen_hvm_init_pci(MachineS
+              pci_type);
+ }
+ 
++extern void pc_q35_init(MachineState *machine);
++
+ static void pc_xen_hvm_init(MachineState *machine)
+ {
+     PCIBus *bus;
+@@ -386,7 +390,10 @@ static void pc_xen_hvm_init(MachineState
+         exit(1);
+     }
+ 
+-    pc_xen_hvm_init_pci(machine);
++    if (!xen_q35) 
++        pc_xen_hvm_init_pci(machine);
++    else 
++        pc_q35_init(machine);
+ 
+     bus = pci_find_primary_bus();
+     if (bus != NULL) {
+@@ -419,7 +426,12 @@ static void pc_i440fx_machine_options(Ma
+ static void pc_i440fx_2_6_machine_options(MachineClass *m)
+ {
+     pc_i440fx_machine_options(m);
+-    m->alias = "pc";
++
++    if (!xen_q35)
++    	m->alias = "pc";
++    else 
++	m->alias = "old-pc";
++
+     m->is_default = 1;
+ }
+ 
+Index: qemu-2.6.2/hw/i386/pc_q35.c
+===================================================================
+--- qemu-2.6.2.orig/hw/i386/pc_q35.c
++++ qemu-2.6.2/hw/i386/pc_q35.c
+@@ -28,6 +28,7 @@
+  * THE SOFTWARE.
+  */
+ #include "qemu/osdep.h"
++#include "sysemu/block-backend.h"
+ #include "hw/hw.h"
+ #include "hw/loader.h"
+ #include "sysemu/arch_init.h"
+@@ -47,12 +48,88 @@
+ #include "hw/usb.h"
+ #include "qemu/error-report.h"
+ #include "migration/migration.h"
++#include "hw/block/flash.h"
++#ifdef CONFIG_XEN
++#  include <xen/hvm/hvm_info_table.h>
++#endif
+ 
+ /* ICH9 AHCI has 6 ports */
+ #define MAX_SATA_PORTS     6
+ 
++extern int xen_q35;
++
++static void xen_map_efi_var_rom(MemoryRegion *rom_memory)
++{
++       DriveInfo *pflash_drv;
++       BlockBackend *blk;
++       int64_t size;
++
++       char *fatal_errmsg = NULL;
++       hwaddr phys_addr = 0xffe00000ULL;
++       int sector_bits, sector_size;
++       char name[64];
++
++        pflash_drv = drive_get(IF_PFLASH, 0, 0);
++
++       if (!pflash_drv) {
++               error_report("XEN EFI NVRAM - No pflash device found\n");
++               exit(1);
++       }
++
++
++               blk = blk_by_legacy_dinfo(pflash_drv);
++               size = blk_getlength(blk);
++
++       sector_bits = 12;
++       sector_size = 1 << sector_bits;
++
++
++        if (size < 0) {
++            fatal_errmsg = g_strdup_printf("failed to get backing file size");
++        } else if (size == 0) {
++            fatal_errmsg = g_strdup_printf("PC system firmware (pflash) "
++                               "cannot have zero size");
```

Comment:
I am all about observing 80 char, and this looks like it would have been fine. Please don't unnecessarily split.